librz/core/cmd_system.c

```diff
@@ -0,0 +1,92 @@
+static inline bool has_quotes(const char *arg, size_t len) {
+	const char *end = arg + len - 1;
+	if (strchr(arg, '"') == arg) {
+		return strchr(arg + 1, '"') == end;
+	}
+	return strchr(arg + 1, '"') == end;
+}
+
+static inline bool has_spaces(const char *arg) {
+	return strchr(arg, ' ') || strchr(arg, '\t');
+}
+
+static char *system_cmd_new(int argc, const char **argv) {
+	if (argc == 1) {
+		return strdup(argv[0]);
+	}
+
+	RzStrBuf *sb = rz_strbuf_new(argv[0]);
+	if (!sb) {
+		return NULL;
+	}
+
+	for (int i = 1; i < argc; ++i) {
+		const char *arg = argv[i];
+		size_t len = strlen(arg);
+		if (len < 1) {
+			rz_strbuf_append(sb, " \"\"");
+			continue;
+		} else if (!has_spaces(arg) || has_quotes(arg, len)) {
+			rz_strbuf_appendf(sb, " %s", arg);
+			continue;
+		}
+		rz_strbuf_append(sb, " \"");
+		if (strchr(arg, '"') || strchr(arg, '\\')) {
+			size_t p = 0, s = 0;
+			for (p = 0, s = 0; p < len; ++p) {
+				if (arg[p] == '\\' || arg[p] == '"') {
+					rz_strbuf_append_n(sb, arg + s, p - s);
+					rz_strbuf_append(sb, "\\");
+					s = p;
+				}
+			}
+			arg += s;
+		}
+		rz_strbuf_appendf(sb, "%s\"", arg);
+	}
```

Comment:
not a big fan of all this logic actually... I *think* @thestr4ng3r already did something like this somewhere. Am I wrong Florian?